### DISCUSSION TITLE
add header for all routes

### DISCUSSION THREAD

**Discussion by sajadshouli** [dedoc/scramble#490] [Q&A]
how can I add some headers for all routes like these headers
`Accept-language`
`timezone`

**Comment by romalytvynenko** [comment]
@sajadshouli Using operation extensions, you can add header params to all of your routes (operations).

**Comment by sajadshouli** [comment]
I read the extension docs but can you give me an example to how add  a header to all routes

**Comment by romalytvynenko** [comment]
@sajadshouli you can follow this example, but add a header instead: https://helgesver.re/articles/operationid-support-scramble-openapi

**Comment by romalytvynenko** [comment]
@sajadshouli and inside the extension you simply write:

```php
$operation->addParameters([
            Parameter::make('Accept-language', 'header'),
            Parameter::make('timezone', 'header'),
        ]);
```

**Comment by sajadshouli** [comment]
thank you @romalytvynenko 

I write this extension

``` php
namespace App\Extensions;

use Dedoc\Scramble\Extensions\OperationExtension;
use Dedoc\Scramble\Support\Generator\Operation;
use Dedoc\Scramble\Support\Generator\Parameter;
use Dedoc\Scramble\Support\RouteInfo;

class AddHeadersToDocsForAdmin extends OperationExtension
{
    public function handle(Operation $operation, RouteInfo $routeInfo)
    {
        $operation->addParameters([
            Parameter::make('Accept-language', 'fa'),
            Parameter::make('timezone', 'Asia/Tehran'),
        ]);
    }
}

```
and add it to config

![image](https://github.com/user-attachments/assets/ed9c3980-b1b1-4ee1-b680-453c94b02336)


but nothing happened in the document

![image](https://github.com/user-attachments/assets/268ddb43-766f-4575-b60c-173446f24c66)

**Reply by romalytvynenko** [reply]
The second parameter you pass to `Parameter::make` **MUST** be `header`. If you want to add an example, you can use something like this:
```php
Parameter::make('Accept-language', 'header')
    ->example('fa')
```

**Reply by sajadshouli** [reply]
is there a function to set a default value directly? the example is good but is some cases I want to make sure the request has a specific header with a specific  value

**Reply by romalytvynenko** [reply]
@sajadshouli https://github.com/dedoc/scramble/discussions/490#discussioncomment-10365223

**Comment by romalytvynenko** [comment]
@sajadshouli 
Try this one
```php
use Dedoc\Scramble\Support\Generator\Parameter;
use Dedoc\Scramble\Support\Generator\Schema;
use Dedoc\Scramble\Support\Generator\Types\StringType;

// ...

Parameter::make('Accept-language', 'header')
    ->example('fa')
    ->setSchema(Schema::fromType(
        (new StringType)->default('fa'),
    ));
```
